torch/csrc/autograd/record_function.h

```diff
@@ -134,6 +134,14 @@ struct TORCH_API RecordFunction {
   // that it was not set with setThreadId() and this RecordFunction's callbacks
   // cannot be invoked from a separate thread.
   uint16_t threadId_ = 0;
+  // Tracks the state of the RecordFunction through its lifetime. Used for error
+  // handling and ensuring that users utilize RecordFunction's functions in the
+  // right way.
+  unsigned int state = 0;
```

Comment:
Any reason for using `unsigned int` instead of `uint32_t`? seems the latter is more clear on how many bits/flags it contains and will be platform-independent?

Reply:
uint8_t should be sufficient?